File: ai-hub/backend/app/api/schemas/conversation.py
"""会话相关 Schema"""

from pydantic import BaseModel


class ConversationCreate(BaseModel):
  """创建会话请求"""
  title: str = "新会话"


class ConversationUpdate(BaseModel):
  """更新会话请求"""
  title: str


class ConversationResponse(BaseModel):
  """会话响应"""
  id: str
  title: str
  created_at: str
  updated_at: str


class MessageResponse(BaseModel):
  """消息响应"""
  id: str
  conversation_id: str
  role: str
  content: str
  metadata: str
  created_at: str

File: ai-hub/frontend/src/shared/types/conversation.ts
/** 会话类型定义 */
export interface Conversation {
  id: string
  title: string
  type?: string          // 'chat' | 'comfort'，后端按此过滤
  createdAt: string
  updatedAt: string
}

export interface MessageData {
  id: string
  conversationId: string
  role: string
  content: string
  metadata: string
  createdAt: string
}
